src/main/resources/static/vulnerableApp.js

```diff
@@ -160,14 +175,20 @@ function handleElementAutoSelection(vulnerableAppEndPointData, id = 0) {
   }
 
   if (id === 0) {
-    detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
-  } else {
+// Assuming detailTitle is a reference to a DOM element
+var detailTitle = document.getElementById('detailTitle');
+
+// Use textContent or createTextNode to safely set the content
+detailTitle.textContent = vulnerableAppEndPointData[id]["Description"];
     innerMaster.innerHTML = "";
   }
 
   vulnerabilitySelected = vulnerableAppEndPointData[id]["Name"];
-  detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities